contracts/eosio.system/include/eosio.system/eosio.system.hpp

```diff
@@ -280,10 +288,126 @@ namespace eosiosystem {
       };
 
       // explicit serialization macro is not necessary, used here only to improve compilation time
-      EOSLIB_SERIALIZE( voter_info, (owner)(proxy)(producers)(staked)(unpaid_voteshare)(unpaid_voteshare_last_updated)(unpaid_voteshare_change_rate)
+      EOSLIB_SERIALIZE( voter_info, (owner)(proxy)(producers)(proposals)(staked)(unpaid_voteshare)(unpaid_voteshare_last_updated)(unpaid_voteshare_change_rate)
                                     (last_claim_time)(last_vote_weight)(proxied_vote_weight)(is_proxy)(flags1)(reserved2)(reserved3) )
    };
 
+    struct [[eosio::table, eosio::contract("eosio.system")]] proposer {
+        name account;
+        string first_name;
+        string last_name;
+        string img_url;
+        string bio;
+        string country;
+        string telegram;
+        string website;
+        string linkedin;
+        time_point_sec last_claim_time;
+        uint64_t primary_key() const { return account.value; }
+        EOSLIB_SERIALIZE( proposer, (account)(first_name)(last_name)(img_url)(bio)(country)(telegram)(website)(linkedin)(last_claim_time) )
+    };
+
+    struct PROPOSAL_STATUS {
```

Comment:
I would suggest to change it for an `enum class`, but it would require some refactor that I cannot fully justify.

Reply:
I appreciate the suggestion, but are there any specific functional / performance differences that this would make?
I opted to use a struct for the proposal status as it's extremely simple and easy for anybody to understand.
If you have a strong preference for one style over the other, feel free to make changes, test, and push on another PR. However, I don't see anything changing in a practical sense. If anything, the code would get longer.

Reply:
Using `enum class` creates a type class type, which is preferred over old style `enum` or defining plain constants:
https://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum

Though, I believe you can face some EOS limitation when you try to save this in a EOS table. If this is the case, you would need to implement some sort of adapter or translator. All this refactor is the one I've mention that I do not justify.